File: tests/fixture-java/src/main/java/com/example/OrderController.java
package com.example;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/orders")
public class OrderController {
    private final OrderService orderService;

    public OrderController(OrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping("/{id}")
    public Object getOrder(@PathVariable Long id) {
        return orderService.findOrder(id);
    }

    @PostMapping
    public Object createOrder(@RequestBody Object body) {
        return orderService.createOrder(body);
    }
}

File: tests/fixture-spring/src/main/kotlin/com/example/order/OrderController.kt
package com.example.order

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/orders")
class OrderController(
    private val orderService: OrderService,
) {
    @PostMapping
    fun placeOrder(@RequestBody req: PlaceOrderRequest): ResponseEntity<OrderDto> {
        return ResponseEntity.ok(orderService.placeOrder(req))
    }

    @DeleteMapping("/{id}")
    fun cancelOrder(@PathVariable id: String): ResponseEntity<Void> {
        orderService.cancelOrder(id)
        return ResponseEntity.noContent().build()
    }

    @GetMapping("/{id}")
    fun getOrder(@PathVariable id: String): ResponseEntity<OrderDto> {
        return ResponseEntity.ok(orderService.findById(id))
    }
}
